src/Logic.hs

```diff
@@ -422,40 +450,53 @@ handleCommentAdded triggerConfig projectConfig mergeWindowExemption prId author
     -- frequently, but most comments are not merge commands, and checking that
     -- a user has push access requires an API call.
     Just pr -> do
-      let approvalType = parseMergeCommand triggerConfig body
+      let commandType = parseMergeCommand triggerConfig body
           exempted :: Username -> Bool
           exempted (Username user) =
             let (Config.MergeWindowExemptionConfiguration users) = mergeWindowExemption
             in elem user users
-      isApproved <-
-        if isJust approvalType
-          then isReviewer author
-          else pure False
+      -- Check whether the author is allowed to do merge commands, but only if
+      -- a valid command was parsed.
+      isAllowed <- if isSuccess commandType
+        then isReviewer author
+        else pure False
       -- For now Friday at UTC+0 is good enough.
       -- See https://github.com/channable/hoff/pull/95 for caveats and improvement ideas.
       day <- dayOfWeek . utctDay <$> getDateTime
-      if isApproved
-        then -- The PR has now been approved by the author of the comment.
-         case fromJust approvalType of
-           -- To guard against accidental merges we make use of a merge window.
-           -- Merging inside this window is discouraged but can be overruled with a special command or by adding the
-           -- user to the merge window exemption list.
-          (approval, _) | exempted author -> handleMergeRequested projectConfig prId author state pr approval
-          (approval, OnFriday) | day == Friday -> handleMergeRequested projectConfig prId author state pr approval
-          (approval, NotFriday)| day /= Friday -> handleMergeRequested projectConfig prId author state pr approval
-          (other, NotFriday) -> do
-            () <- leaveComment prId ("Your merge request has been denied, because \
-                                      \merging on Fridays is not recommended. \
-                                      \To override this behaviour use the command `"
-                                      <> Pr.displayApproval other <> " on Friday`.")
-            pure state
-          (other, OnFriday) -> do
-            () <- leaveComment prId ("Your merge request has been denied because \
-                                      \it is not Friday. Run " <>
-                                      Pr.displayApproval other <> " instead")
-            pure state
-        else pure state
-     -- If the pull request is not in the state, ignore the comment.
+      case commandType of
+        -- The bot was not mentioned in the comment, ignore
+        Ignored -> pure state
+        -- The bot was mentioned but encountered an invalid command, report error and
+        -- take no further action
+        Unknown command -> do
+          let prefix = Text.toCaseFold $ Config.commentPrefix triggerConfig
+              cmdstr = Text.strip $ fromJust $ Text.stripPrefix prefix command
```

Comment:
In general, I don't like `fromJust`, the error message is quite terrible. I know it has been used quite a lot in Hoff, but let's not make it worse. Can you replace it with `fromMaybe (error "Some custom error message) $ `